packages/aws-cdk-lib/aws-events-targets/test/appsync/appsync.test.ts

```diff
@@ -0,0 +1,394 @@
+    });
+
+    const rule = new events.Rule(stack, 'Rule', {
+      schedule: events.Schedule.rate(cdk.Duration.minutes(1)),
+    });
+
+    expect(() => {
+      rule.addTarget(new targets.AppSync(noiam_api, {
+        graphQLOperation,
+        variables: events.RuleTargetInput.fromObject({
+          message: events.EventField.fromPath('$.detail'),
+        }),
+      }));
+    }).toThrow('You must have AWS_IAM authorization mode enabled on your API to configure an AppSync target');
+  });
+
+  test('fails when graphQLEndpointArn is not configured', () => {
+    const noEndpointArnAPI = appsync.GraphqlApi.fromGraphqlApiAttributes(stack, 'ImportedAPI', {
+      graphqlApiId: 'MyApiId',
+      graphqlApiArn: 'MyApiArn',
+    });
+
+    const rule = new events.Rule(stack, 'Rule', {
+      schedule: events.Schedule.rate(cdk.Duration.minutes(1)),
+    });
+
+    expect(() => {
+      rule.addTarget(new targets.AppSync(noEndpointArnAPI, {
+        graphQLOperation,
+        variables: events.RuleTargetInput.fromObject({
+          message: events.EventField.fromPath('$.detail'),
+        }),
+      }));
+    }).toThrow('You must have AWS_IAM authorization mode enabled on your API to configure an AppSync target');
+  });
+
+  test('Accepts API create with fromGraphqlApiAttributes', () => {
+    const api = appsync.GraphqlApi.fromGraphqlApiAttributes(stack, 'ImportedAPI', {
+      graphqlApiId: 'MyApiId',
+      graphqlApiArn: 'MyApiArn',
+      graphQLEndpointArn: 'arn:aws:appsync:us-east-2:000000000000:endpoints/graphql-api/00000000000000000000000000',
+      visibility: appsync.Visibility.GLOBAL,
+      modes: [appsync.AuthorizationType.IAM],
+    });
+
+    const rule = new events.Rule(stack, 'Rule', {
+      schedule: events.Schedule.rate(cdk.Duration.minutes(1)),
+    });
+
+    // WHEN
+    rule.addTarget(new targets.AppSync(api, {
+      graphQLOperation,
+      variables: events.RuleTargetInput.fromObject({
+        message: events.EventField.fromPath('$.detail'),
+      }),
+    }));
+
+    // THEN
+    Template.fromStack(stack).hasResourceProperties('AWS::Events::Rule', {
+      Targets: [
+        {
+          Arn: api.graphQLEndpointArn,
+          AppSyncParameters: { GraphQLOperation: graphQLOperation },
+          Id: 'Target0',
+          InputTransformer: {
+            InputPathsMap: { detail: '$.detail' },
+            InputTemplate: '{"message":<detail>}',
+          },
+          RoleArn: {
+            'Fn::GetAtt': [
+              'ImportedAPIEventsRole9CE171B7',
+              'Arn',
+            ],
+          },
+        },
+      ],
+    });
+  });
+
+  test('allows secondary auth with AWS_IAM configured', () => {
+    const sec_api = new appsync.GraphqlApi(stack, 'sec_api', {
+      name: 'no_iam_api',
+      definition: appsync.Definition.fromFile(path.join(__dirname, 'appsync.test.graphql')),
+      authorizationConfig: { additionalAuthorizationModes: [{ authorizationType: appsync.AuthorizationType.IAM }] },
```

Comment:
what happens if the secondary auth type is not `IAM` ?

Reply:
`IAM` needs to be one of the supported modes. either primary or secondary. if none are configured as `IAM` the target definition should fail.